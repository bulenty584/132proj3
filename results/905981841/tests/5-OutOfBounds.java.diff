1,17c1,3
< Exception in thread "main" ParseException: Encountered " <INTEGER_LITERAL> "1 "" at line 22, column 17.
< Was expecting one of:
<     <IDENTIFIER> ...
<     <IDENTIFIER> ...
<     <IDENTIFIER> ...
<     <IDENTIFIER> ...
<     <IDENTIFIER> ...
<     
< 	at SparrowParser.generateParseException(SparrowParser.java:1202)
< 	at SparrowParser.jj_consume_token(SparrowParser.java:1084)
< 	at SparrowParser.Identifier(SparrowParser.java:506)
< 	at SparrowParser.Add(SparrowParser.java:224)
< 	at SparrowParser.Instruction(SparrowParser.java:123)
< 	at SparrowParser.Block(SparrowParser.java:84)
< 	at SparrowParser.FunctionDeclaration(SparrowParser.java:66)
< 	at SparrowParser.Program(SparrowParser.java:23)
< 	at Main.main(Main.java:24)
---
> 0
> --- Error (Explicit) ---
> array index out of bounds
