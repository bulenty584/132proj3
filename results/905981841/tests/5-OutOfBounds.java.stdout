func Main()
  y0 = 0
  z0 = 4
  z1 = alloc(z0)
  z2 = 4
  vmt_A = alloc(z2)
  z3 = @A_run
  [vmt_A + 0] = z3
  [z1 + 0] = vmt_A
  if0 z1 goto z1Error_0
  goto z1End_1
z1Error_0:
  error("null pointer")
z1End_1:
  if0 z1 goto z1Error_2
  goto z1End_3
z1Error_2:
  error("null pointer")
z1End_3:
  z4 = [z1 + 0]
  z5 = [z4 + 0]
  z6 = call z5(z1)
  print(z6)
  return y0
func A_run(this)
  z7 = 20
  z9 = 1
  z8 = z7 + z9
  z11 = 4
  z10 = z8 * z11
  z12 = alloc(z10)
  if0 z12 goto z12Error_4
  goto z12End_5
z12Error_4:
  error("null pointer")
z12End_5:
  [z12 + 0] = z7
  a = z12
  z13 = 10
  z14 = [a + 0]
  z15 = 4
  z16 = 1
  z17 = 0
  z18 = z17 - z16
  z19 = z18 < z13
  z20 = z13 < z14
  z21 = z19 * z20
  if0 z21 goto z21Error_6
  goto z21End_7
z21Error_6:
  error("array index out of bounds")
z21End_7:
  z22 = z13 * z15
  z23 = z22 + z15
  z24 = a + z23
  z25 = [z24 + 0]
  print(z25)
  z26 = 40
  z27 = [a + 0]
  z28 = 4
  z29 = 1
  z30 = 0
  z31 = z30 - z29
  z32 = z31 < z26
  z33 = z26 < z27
  z34 = z32 * z33
  if0 z34 goto z34Error_8
  goto z34End_9
z34Error_8:
  error("array index out of bounds")
z34End_9:
  z35 = z26 * z28
  z36 = z35 + z28
  z37 = a + z36
  z38 = [z37 + 0]
  return z38
